src/core/network/createResponse.js

```diff
@@ -25,23 +25,25 @@ import { assign, find } from "../../utils";
  * }
  *
  * @returns {Object<Response>} A Response object containing:
- *  - All the properties of the raw response, frozen
  *  - `getPayloadByType`: returns a fragment of the response by type
  *      - @param {String} type: A string with the current format: <namespace:action>
  *          example: "identity:persist"
  */
-export default (respDto = { requestId: "", handle: [] }) => {
-  const response = assign(Object.create(null), respDto);
+export default (content = { requestId: "", handle: [] }) => {
   // TODO: Should we freeze the response to prevent change by Components?
   // Object.freeze(response.handle.map(h => Object.freeze(h)));
-
-  response.getPayloadByType = type => {
-    const fragment = find(respDto.handle, content => content.type === type);
-    return fragment ? fragment.payload : null;
+  return {
+    getPayloadByType(type) {
+      const fragment = find(
+        content.handle,
+        handleContent => handleContent.type === type
+      );
+      return fragment ? fragment.payload : null;
+    },
+    toJSON() {
+      return content;
+    }
+    // TODO: Maybe consider the following API as well?
+    // - getPayloadsByAction
   };
```

Comment:
The way `createResponse.js` was set up was different from `createPayload.js` and it was useful to make them more consistent. That is, both of them have accessors to get underlying data and both of them have `toJSON` defined to get the underlying data for serialization and cloning purposes. In the case of this pull request, this allows to us to clone the response and provide the result to users so that any changes they make to that data will not affect the data we hold internally.